src/main.rs

```diff
@@ -14,8 +14,14 @@ pub fn main() {
     session.add_event_start(ActivityKind::Notion);
 
     let notion = cli::Notion::from_args();
+    let verbose = notion.verbose;
     let exit_code = notion.run(&mut session).unwrap_or_else(|err| {
-        display_error(ErrorContext::Notion, &err);
+        let reporter = if verbose {
```

Comment:
I find myself kinda wanting to see this logic get folded into a single constructor method for `ErrorReporter` so here you could write:

```rust
ErrorReporter::from_flag(notion.verbose).report(ErrorContext::Notion, &err);
```

And you could rename the existing `new` constructor `from_env()`, and `from_flag(false)` would call that.

Just a thought!

Reply:
That's a nice approach, I'll do that. I was trying to avoid passing a boolean in, but still having 2 constructors just with better names makes a lot more sense. My main issue was I didn't want to do `ErrorReporter::new(false)` in `shim.rs`, because magic boolean values are difficult to maintain.